test/framework/integration_framework/fake_peer/fake_peer.cpp

```diff
@@ -119,6 +130,7 @@ namespace integration_framework {
 
     FakePeer &FakePeer::setBlockStorage(
         const std::shared_ptr<BlockStorage> &block_storage) {
+      ensureInitialized();
```

Comment:
Why we can't use `initialize()` here instead `ensureInitialized()`?

Can we put such check into `initialize()`?
```diff
    FakePeer &FakePeer::initialize() {
-      BOOST_VERIFY_MSG(!initialized_, "Already initialized!"); 
+      if (not initialized_) {
        // here comes the initialization of members requiring shared_from_this()
        synchronizer_transport_ =
            std::make_shared<LoaderGrpc>(shared_from_this());

        initialized_ = true;
+      }
      return *this;
    }
```

Reply:
I would like to force initialization at creation time before the object starts being used. That is why there is an assertion.